### PR TITLE
Parse .aud file metadata correctly

### PR DESCRIPTION
**Pull request by Mailaender**
This is a followup of https://github.com/OpenRA/OpenRA/pull/18911.

### CODE REVIEW

File: OpenRA.Mods.Cnc/FileFormats/AudReader.cs [hunk 1]
```diff
@@ -59,14 +60,23 @@ public static bool LoadSound(Stream s, out Func<Stream> result, out int sampleRa
...
 				var outputSize = s.ReadInt32();
 
+				sampleBits = 0;
+				channels = 0;
+
 				var readFlag = s.ReadByte();
 				if (!Enum.IsDefined(typeof(SoundFlags), readFlag))
 					return false;
 
+				sampleBits = readFlag == (int)SoundFlags._8Bit ? 8 : 16;
```
Comment (pchote) [inline OpenRA.Mods.Cnc/FileFormats/AudReader.cs]:
```
sampleBits = (readFlag & (int)SoundFlags._16Bit) == 0 ? 8 : 16;
channels = (readFlag & (int)SoundFlags.Stereo) == 0 ? 1 : 2;
```
and then delete the `_8Bit` definition. These are flags, not magic values!

Comment (Mailaender) [reply OpenRA.Mods.Cnc/FileFormats/AudReader.cs]: That explains why I didn't find that definition in the original file headers.

File: OpenRA.Mods.Cnc/AudioLoaders/AudLoader.cs [hunk 2]
```diff
@@ -25,7 +25,7 @@ bool IsAud(Stream s)
 			var readFormat = s.ReadByte();
 			s.Position = start;
 
-			if (!Enum.IsDefined(typeof(SoundFlags), readFlag))
+			if (readFlag > (int)SoundFlags._16Bit)
```
Comment (pchote) [inline OpenRA.Mods.Cnc/AudioLoaders/AudLoader.cs]: These checks should match the actual file loading, so this should be removed and a new WestwoodCompressed check added after the SoundFormat check below.

Comment (Mailaender) [reply OpenRA.Mods.Cnc/AudioLoaders/AudLoader.cs]: That won't change anything, though. There is already a `try catch` which returns false.

Comment (pchote) [reply OpenRA.Mods.Cnc/AudioLoaders/AudLoader.cs]: This is about code maintainability rather than code logic.

**Review by abcdefg30 — APPROVED** [review]
Code changes lgtm, the sounds from our downloadable packages work and I still get `nuyell1.aud is not a valid sound file!` when trying to play an unsupported aud. :+1:

**Review by pchote — APPROVED** [review]
I still think that these would be better PRed as commits within a larger PR that actually finishes the feature, but ok.

**Comment by pchote** [comment]
https://github.com/OpenRA/OpenRA/wiki/Changelog-(bleed)/_compare/2c3b40dc3333660a2d9ef66cce5b28468161e6f5

**Comment by Mailaender** [comment]
https://github.com/Mailaender/OpenRA/tree/aud-8 is on hold. I have no idea to fix it.

**Pull request merged**